src/main/java/net/starlark/java/cmd/Main.java

```diff
@@ -72,14 +76,31 @@ private static String prompt() {
         }
         input.append(lineSeparator).append(line);
         lineSeparator = "\n";
-        System.out.print(CONTINUATION_PROMPT);
+        prompt = CONTINUATION_PROMPT;
       }
-    } catch (IOException e) {
-      System.err.format("Error reading line: %s\n", e);
+    } catch (UserInterruptException | EndOfFileException e) {
       return null;
     }
   }
 
+  private static LineReader buildLineReader() {
+    try {
+      Terminal terminal = TerminalBuilder.terminal();
+      LineReaderBuilder builder = LineReaderBuilder.builder().terminal(terminal);
+      builder.terminal(terminal);
+      // By default jline moves cursor for a short time to the matching paren
+      builder.variable(LineReader.BLINK_MATCHING_PAREN, false);
+      String userHome = System.getProperty("user.home");
+      if (userHome != null) {
+        // Persist history by default
+        builder.variable(LineReader.HISTORY_FILE, userHome + "/.starlark_java_history");
+      }
+      return builder.build();
+    } catch (IOException e) {
+      throw new RuntimeException(e);
```

Comment:
Throwing RuntimeException is never correct. Either this "can't happen", in which case throw IllegalStateException, or it can happen, in which case throw a checked exception and handle it in the caller. I assume it's the latter in this instance, but I'm not familiar with the jline API or its failure modes.